torch/_subclasses/meta_utils.py

```diff
@@ -394,12 +398,48 @@ def is_c_of_r(complex_dtype, real_dtype):
 
                 else:
                     is_leaf = safe_is_leaf(t)
-                    sizes, strides, storage_offset = sym_sizes_strides_storage_offset(t)
-                    r = callback(
-                        lambda: torch.empty_strided(
-                            sizes, strides, dtype=t.dtype, device="meta"
-                        )
+                    sizes, strides, storage_offset = sym_sizes_strides_storage_offset(
+                        t, source
                     )
+
+                    # If we have a subclass that desugars into dense tensors,
+                    # perform our callback on each inner tensor.
+                    if is_traceable_wrapper_subclass(t):
+
+                        def empty_create(inner_t, inner_src):
+                            is_leaf = safe_is_leaf(inner_t)
```

Comment:
looks dead?

Reply:
yep (we don't need autograd meta to be correct on the inner tensors anyway)